pkg/gardenlet/controller/networkpolicy/controller.go

```diff
@@ -58,8 +60,17 @@ type Controller struct {
 	waitGroup               sync.WaitGroup
 }
 
-// NewController instantiates a new controller.
-func NewController(ctx context.Context, seedClient kubernetes.Interface, seedDefaultNamespaceKubeInformer kubeinformers.SharedInformerFactory, seedLogger *logrus.Entry, recorder record.EventRecorder, seedName string) (*Controller, error) {
+// NewController instantiates a new networkpolicy controller.
+func NewController(ctx context.Context, clientMap clientmap.ClientMap, logger *logrus.Logger, recorder record.EventRecorder, seedName string) (*Controller, error) {
+	seedClient, err := clientMap.GetClient(ctx, keys.ForSeedWithName(seedName))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get garden client: %w", err)
+	}
```

Comment:
You can pass the `seedClient` directly in `factory.go`, i.e., no need to pass the entire `clientMap`.